hlx_statics/scripts.js

```diff
@@ -1,17 +1,46 @@
 let $IS_HLX_PATH = false;
 let $IS_STAGE = false;
 
-if(window.location.host.indexOf('hlx.page') >= 0 || window.location.host.indexOf('hlx.live') >= 0 || window.location.host.indexOf('localhost') >= 0){
+if (window.location.host.indexOf('hlx.page') >= 0 || window.location.host.indexOf('hlx.live') >= 0 || window.location.host.indexOf('localhost') >= 0) {
   $IS_HLX_PATH = true;
 }
 
-if(window.location.host.indexOf('stage.adobe.io') >= 0 || window.location.host.indexOf('developer-stage') >= 0){
+if (window.location.host.indexOf('stage.adobe.io') >= 0 || window.location.host.indexOf('developer-stage') >= 0) {
   $IS_STAGE = true;
 }
 
 window.adobeid = {};
 
-if($IS_HLX_PATH) {
+// Used to update the url in the browser
+const setQueryStringParameter = (name, value) => {
+  const params = new URLSearchParams(window.location.search);
+  params.set(name, value);
+  window.history.replaceState({}, "", `${window.location.pathname}?${params}`);
+};
+const getQueryStringParameter = (name) => {
+  const params = new URLSearchParams(window.location.search);
+  return params.get(name);
+};
+const getQueryString = () => {
+  const params = new URLSearchParams(window.location.search);
+  return params.toString();
+}
+
+window.addEventListener('message', function (e) {
+
+  const expectedOrigin = window.location.hostname === 'localhost' ? 'http://localhost:8000' : window.location.hostname === 'developer.adobe.com' ? 'https://developer.adobe.com' : 'https://developer-stage.adobe.com';
```

Comment:
Following up on slack conversation with @timkim I will be refactoring into if/else conditional to improve readability.

Reply:
done